include/deal.II/lac/vector.h

```diff
@@ -967,12 +967,9 @@ class Vector : public Subscriptor
 
 
 //------------------------ declarations for explicit specializations
-
-#ifdef DEAL_II_BOOST_BIND_COMPILER_BUG
 template <>
 Vector<std::complex<float> > &
 Vector<std::complex<float> >::operator= (const std::complex<float> s);
-#endif
```

Comment:
If `DEAL_II_BOOST_BIND_COMPILER_BUG` is never defined (because it is a very old bug), why not simply removing lines 970 - 972?